internal/v6provider/v6provider.go

```diff
@@ -0,0 +1,341 @@
+` + "`KATAPULT_SKIP_TRASH_OBJECT_PURGE`" + ` environment variable. Defaults to ` + "`false`" + `.
+
+  ~> **Note:** Using ` + "`skip_trash_object_purge`" + ` can quickly lead to a build up of a lot objects in the trash if you are replacing resources repeatedly. Hence this option is disabled by default, and should only be used if you are sure you want to keep deleted resources in the trash.
+
+`),
+				//nolint:lll
+				MarkdownDescription: strings.TrimSpace(`
+
+Skip purging deleted resources from Katapult's trash when they are destroyed by Terraform. Only relevant to some resources which are moved to the trash when they are deleted. Can be specified with the
+` + "`KATAPULT_SKIP_TRASH_OBJECT_PURGE`" + ` environment variable. Defaults to ` + "`false`" + `.
+
+  ~> **Note:** Using ` + "`skip_trash_object_purge`" + ` can quickly lead to a build up of a lot objects in the trash if you are replacing resources repeatedly. Hence this option is disabled by default, and should only be used if you are sure you want to keep deleted resources in the trash.
+
+`),
+			},
+			"log_level": schema.StringAttribute{
+				Optional: true,
+
+				Validators: []validator.String{
+					stringvalidator.OneOfCaseInsensitive("trace",
+						"debug",
+						"info",
+						"warn",
+						"error",
+						"off",
+					),
+				},
+				Description: "Log level used by Katapult Terraform " +
+					"provider. Can be specified with the " +
+					"`KATAPULT_LOG_LEVEL` environment variable. " +
+					"Defaults to `info`.",
+				MarkdownDescription: "Log level used by Katapult Terraform " +
+					"provider. Can be specified with the " +
+					"`KATAPULT_LOG_LEVEL` environment variable. " +
+					"Defaults to `info`.",
+			},
+		},
+	}
+}
+
+func stringOrEnv(in string, env string) string {
+	if in != "" {
+		return in
+	}
+
+	return os.Getenv(env)
+}
+
+func boolOrEnv(in *bool, env string) bool {
+	if in != nil {
+		return *in
+	}
+
+	switch strings.ToLower(os.Getenv(env)) {
+	case "true", "1":
+		return true
+	}
+
+	return false
+}
+
+func userAgent(name string, terraformVersion string, version string) string {
+	ua := fmt.Sprintf(
+		"Terraform/%s (+https://www.terraform.io) Terraform-Plugin-Framework",
+		terraformVersion,
+	)
+	if name != "" {
+		ua += " " + name
+		if version != "" {
+			ua += "/" + version
+		}
+	}
+
+	if add := os.Getenv(uaEnvVar); add != "" {
+		add = strings.TrimSpace(add)
+		if len(add) > 0 {
+			ua += " " + add
+			log.Printf("[DEBUG] Using modified User-Agent: %s", ua)
+		}
+	}
+
+	return ua
+}
+
+//nolint:funlen // just over 100 lines
+func (k *KatapultProvider) Configure(
+	ctx context.Context,
+	req provider.ConfigureRequest,
+	resp *provider.ConfigureResponse,
+) {
+	if k.m != nil {
+		resp.ResourceData = k.m
+		resp.DataSourceData = k.m
+		return
+	}
+
+	var conf KatapultProviderModel
+	diags := req.Config.Get(ctx, &conf)
+	resp.Diagnostics.Append(diags...)
+
+	m := &Meta{
```

Comment:
Could we use the `NewMeta()` function here? Or is it more specifically tuned to the test setup?